File: ROS/catkin_ws_basic/src/ros_python_example/ros_python_scripts/srv_server.py
from __future__ import print_function

from std_srvs.srv import Trigger, TriggerResponse, TriggerRequest
import rospy

def generate(request):
    print("Enter generate...")
    return TriggerResponse(success=True)

def add_two_ints_server():
    rospy.init_node('my_ros_python_server')
    s = rospy.Service('generate_random_center', Trigger, generate)
    print("Ready to generate_random_center")
    rospy.spin()


if __name__ == "__main__":
    add_two_ints_server()

File: ROS/catkin_ws_basic/src/ros_misc_test/src/srv_server.cpp
#include <ros/ros.h>
#include <std_srvs/Trigger.h>

bool Generate(std_srvs::TriggerRequest& req,
              std_srvs::TriggerResponse& res)
{
    ROS_INFO("Has a request...");
    res.success = true;
    res.message = "zhanghaiming";
    return true;
}

int main(int argc, char** argv) {
    ros::init(argc, argv, "my_ros_server");
    ros::NodeHandle nh;

    ros::ServiceServer service = nh.advertiseService("generate_random_center", Generate);
    ros::spin();

    return 0;
}